test/normal/apply_test.go

```diff
@@ -0,0 +1,58 @@
+//  Copyright 2021 Jake Son
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package test
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+)
+
+func TestApplySingle(t *testing.T) {
+	applied := deriveApplyMarshal(json.Marshal, map[string]int{"value": 10})
+	want := `{"value":10}`
+	got, err := applied()
```

Comment:
Oh very interesting.
So apply single, turns into lazy apply.
Yeah that works for me :)